src/Data/Fin/Relation/Unary/Top.agda

```diff
@@ -0,0 +1,141 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The 'top' view of Fin
+--
+-- This is an example of a view of (elements of) a datatype,
+-- here i : Fin (suc n), which exhibits every such i as
+-- * either, i = fromℕ n
+-- * or, i = inject₁ j for a unique j : Fin n
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Fin.Relation.Unary.Top where
+
+open import Data.Empty using (⊥; ⊥-elim)
+open import Data.Nat.Base using (ℕ; zero; suc; _<_; _∸_)
+open import Data.Nat.Properties using (n∸n≡0; +-∸-assoc)
+open import Data.Product using (uncurry)
+open import Data.Sum using (inj₁; inj₂)
+open import Data.Fin.Base hiding (_<_)
+open import Data.Fin.Properties as Fin
+  using (suc-injective; toℕ-fromℕ; toℕ<n; toℕ-inject₁; inject₁ℕ<)
+open import Relation.Binary.PropositionalEquality
+
+private
+  variable
+    n : ℕ
+
+------------------------------------------------------------------------
+-- The View, considered as a unary relation on Fin (suc n)
+
+-- First, a lemma not in `Data.Fin.Properties`,
+-- but which establishes disjointness of the
+-- (interpretations of the) constructors of the View
+
+top≢inject₁ : ∀ (j : Fin n) → fromℕ n ≢ inject₁ j
+top≢inject₁ (suc j) eq = top≢inject₁ j (suc-injective eq)
+
+-- The View, itself
+
+data View : (i : Fin (suc n)) → Set where
+
+  top :                View (fromℕ n)
+  inj : (j : Fin n) → View (inject₁ j)
+
+-- The view covering function, witnessing soundness of the view
+
+view : ∀ {n} i → View {n} i
+view {n = zero}  zero = top
+view {n = suc _} zero = inj _
+view {n = suc n} (suc i) with view {n} i
+... | top   = top
+... | inj j = inj (suc j)
+
+-- Interpretation of the view constructors
+
+⟦_⟧ : ∀ {i} → View {n} i → Fin (suc n)
+⟦ top ⟧   = fromℕ _
+⟦ inj j ⟧ = inject₁ j
+
+-- Completeness of the view
+
+view-complete : ∀ {i} (v : View {n} i) → ⟦ v ⟧ ≡ i
+view-complete top     = refl
+view-complete (inj j) = refl
+
+-- 'Computational' behaviour of the covering function
+
+view-top : ∀ n → view {n} (fromℕ n) ≡ top
+view-top zero    = refl
+view-top (suc n) rewrite view-top n = refl
+
+view-inj : ∀ j → view {n} (inject₁ j) ≡ inj j
+view-inj zero       = refl
+view-inj (suc j) rewrite view-inj j = refl
+
+------------------------------------------------------------------------
+-- Experimental
+--
+-- Because we work --without-K, Agda's unifier will complain about
```

Comment:
This comment is maybe behind the fashions of the library. I think replacing "`--without-K`" with the more prosey "without K" should be good enough